_posts/2016-04-30-akka-basics.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+name: Akka-basics
+title: 'Akka basics'
+date: 2016-04-30 00:20:00 +00:00
+author: Felipe Fernández
+image:
+    src: /assets/img/custom/blog/akka.jpg
+    attribution:
+      text: Akka_20120331 by Helena Whitlock, SGU
+      href: https://www.flickr.com/photos/geologicalsurveyofsweden/8703286181
+tags:
+- akka
+- OOP
+- functional programming
+
+---
+
+As [the previous post](http://codurance.com/2016/04/28/async-systems-with-sync-clients) explained our system is based on [Akka](http://akka.io/). Before going into more detail about our solution, I'd like to explain what Akka is about and why it is so awesome.
+
+Akka is a toolkit and not a framework, you can simply use the bits that you need for your service. In this series we'll focus on the core features and we won't go through [Akka Streams](http://doc.akka.io/docs/akka/2.4.4/scala/stream/index.html), [Akka-Http](http://doc.akka.io/docs/akka/2.4.4/scala/http/index.html) or [Akka Cluster](http://doc.akka.io/docs/akka/2.4.4/common/cluster.html) (since they're out of scope, not because they're not fantastic).
+
+Akka provides a different abstraction to deal with concurrency, paralellism and fault-tolerance. That abstraction is called Actor Model. If you have experience with old Java ecosystem you will know how hard [writing safe and correct multithreaded code](http://codurance.com/2015/12/13/testing-multithreaded-code-in-java/) is. Providing a human-friendly abstractions like Actors it's a revolution similar to what Java did with memory management. New challenges derived from internet scale and cloud computing requires a reactive approach to programming. Akka aligns with that philosophy and enables you to implement [Event-driven architectures](https://www.wikiwand.com/en/Event-driven_architecture).
```

Comment:
`you will know how hard it is to [write safe and correct multithreaded code](http://codurance.com/2015/12/13/testing-multithreaded-code-in-java/).`